onyo/lib/git.py

```diff
@@ -401,3 +404,85 @@ def check_ignore(self, ignore: Path, paths: list[Path]) -> list[Path]:
 
     # TODO: git check-ignore --no-index --stdin (or via command call)  ->  lazy, check GitRepo.files once. (Same invalidation)
     #       -> OnyoRepo would use it to maintain a ignore list from a (top-level .onyoignore)? .onyo/ignore ? Both?
+
+    def _parse_log_output(self, lines: list[str]) -> dict:
+        """Generate a dict from the output of ``git log`` for one commit.
+
+        Internal helper that gets a list of git-log-output lines,
+        that represent a single commit and generates a dictionary from it.
+        """
+        import datetime
+        import re
+        regex_author = re.compile(r"Author:\s+(?P<name>\b.*\b)\s+<(?P<email>[^\s]+)>$")
+        regex_committer = re.compile(r"Commit:\s+(?P<name>\b.*\b)\s+<(?P<email>[^\s]+)>$")
+        commit = dict()
+        for line in lines:
+            if line.startswith('commit '):
+                commit['commit'] = line.split()[1]
+                continue
+            elif line.startswith('Author:'):
+                try:
+                    commit['author'] = re.match(regex_author, line).groupdict()  # pyre-ignore [16] AttributeError is caught
+                except AttributeError as e:
+                    if str(e).endswith("'groupdict'"):
+                        raise RuntimeError(f"Unable to parse author-line:\n{line}") from e
+                    raise
+                continue
+            elif line.startswith('AuthorDate:'):
+                continue
+            elif line.startswith('Commit:'):
+                try:
+                    commit['committer'] = re.match(regex_committer, line).groupdict()
+                except AttributeError as e:
+                    if str(e).endswith("'groupdict'"):
+                        raise RuntimeError(f"Unable to parse committer-line:\n{line}") from e
+                    raise
+                continue
+            elif line.startswith('CommitDate:'):
+                commit['time'] = datetime.datetime.fromisoformat(line.split(maxsplit=1)[1])
+                continue
+            else:
+                # This line is part of the message.
+                if 'message' not in commit:
+                    commit['message'] = [line]
+                else:
+                    commit['message'].append(line)
+
+        return commit
+
+    def follow(self, path: Path | None = None, n: int | None = None) -> Generator[dict, None, None]:
```

Comment:
Not thrilled about the name...

What about `history()` or `log()` --- and possibly make `follow` a boolean rather than an implicit behavior of `path`?

Reply:
Agreed. Started w/o `path` being optional and never changed the name.